homeassistant/components/powerwall/entity.py

```diff
@@ -43,3 +43,36 @@ def __init__(self, powerwall_data: PowerwallRuntimeData) -> None:
     def data(self) -> PowerwallData:
         """Return the coordinator data."""
         return self.coordinator.data
+
+
+class BatteryEntity(CoordinatorEntity[DataUpdateCoordinator[PowerwallData]]):
+    """Base class for battery entities."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self, powerwall_data: PowerwallRuntimeData, battery: BatteryResponse
```

Comment:
It would be so much better if PowerwallRuntimeData was a dataclass as we wouldn't have to use as many type hints.  This integration predates use being able to use dataclasses so it's never been converted.